--- irrlicht/source/Irrlicht/Android/jni/Android.mk.orig	2016-02-15 05:33:03.000000000 +0200
+++ irrlicht/source/Irrlicht/Android/jni/Android.mk	2016-02-15 05:34:34.913711815 +0200
@@ -11,7 +11,8 @@
 ifndef NDEBUG
 LOCAL_CFLAGS += -g -D_DEBUG
 else
-LOCAL_CFLAGS += -fexpensive-optimizations -O3
+LOCAL_CFLAGS += -mfpu=vfpv3-d16 -D_NDK_MATH_NO_SOFTFP=1 -mfloat-abi=hard -march=armv7-a -Ofast -fdata-sections -ffunction-sections -fmodulo-sched -fmodulo-sched-allow-regmoves
+LOCAL_LDFLAGS = -Wl,--no-warn-mismatch,--gc-sections -lm_hard
 endif
 
 LOCAL_C_INCLUDES := ../../../include
