--- irrlicht/source/Irrlicht/Android/jni/Android.mk.orig	2016-02-15 05:33:03.000000000 +0200
+++ irrlicht/source/Irrlicht/Android/jni/Android.mk	2016-02-15 05:34:34.913711815 +0200
@@ -11,7 +11,15 @@
 ifndef NDEBUG
 LOCAL_CFLAGS += -g -D_DEBUG
 else
-LOCAL_CFLAGS += -fexpensive-optimizations -O3
+ifeq ($(TARGET_ABI),armeabi-v7a)
+LOCAL_CFLAGS   += -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16 -mthumb
+endif
+ifeq ($(TARGET_ABI),x86)
+LOCAL_CFLAGS   += -march=i686 -mtune=intel -mssse3 -mfpmath=sse -m32 -funroll-loops
+endif
+LOCAL_CFLAGS		+= -Ofast -fdata-sections -ffunction-sections -fvisibility=hidden -flto
+LOCAL_CXXFLAGS := $(LOCAL_CFLAGS)
+LOCAL_LDFLAGS  := -Wl,--no-warn-mismatch,--gc-sections,--icf=safe
 endif

 LOCAL_C_INCLUDES := ../../../include
